File: javascript/easy/perfectNumber.js
/**
 * @param {number} num
 * @return {boolean}
 */
var checkPerfectNumber = function(num) {
    if(num <= 1) return false;
    let sum = 1, sqrt = parseInt(Math.sqrt(num));
    for(let i = 2; i <= sqrt; i++) {
        if((num % i) === 0) {
            let div = parseInt(num/i);
            if(i <= div) {
                sum += (i + div);
            }
        }
    }
    return sum === num;
};

File: ruby/easy/perfect_number.rb
# @param {Integer} num
# @return {Boolean}
def check_perfect_number(num)
    return false if num <= 1
    sum, sqrt = 1, Math.sqrt(num)
    for i in 2..sqrt
        if (num % i) == 0 && i <= (num/i)
            sum += i + (num/i)
        end
    end
    return sum == num
end
